pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -52,25 +54,30 @@ func (pg *pgcontroller) addPod(obj interface{}) {
 	pg.queue.Add(req)
 }
 
+func escapeJSONPointer(p string) string {
+	// Escaping reference name using https://tools.ietf.org/html/rfc6901
+	p = strings.Replace(p, "~", "~0", -1)
+	p = strings.Replace(p, "/", "~1", -1)
+	return p
+}
+
 func (pg *pgcontroller) updatePodAnnotations(pod *v1.Pod, pgName string) error {
 	if pod.Annotations == nil {
 		pod.Annotations = make(map[string]string)
 	}
 	if pod.Annotations[scheduling.KubeGroupNameAnnotationKey] == "" {
-		pod.Annotations[scheduling.KubeGroupNameAnnotationKey] = pgName
+		patch := fmt.Sprintf(`[{"op": "add", "path": "/metadata/annotations/%s", "value":"%s"}]`,
+			escapeJSONPointer(scheduling.KubeGroupNameAnnotationKey), pgName)
+		if _, err := pg.kubeClient.CoreV1().Pods(pod.Namespace).Patch(context.TODO(), pod.Name, types.JSONPatchType, []byte(patch), metav1.PatchOptions{}); err != nil {
```

Comment:
need to format the code

Reply:
Updated, use other formal way to patch pod annotation.